### DISCUSSION TITLE
Bat file issue after converting to 7.1.1-Q16 _ PHP content...

### DISCUSSION THREAD

**Discussion by candidsounds** [ImageMagick/ImageMagick#7397] [Help] [answered]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

Windows 10 pro, PHP

### Description

Hi,  I have been running a PHP script to read images in a local folder that I select manually (in PHP).  The PHP generates a BAT file that calls Magick and tells it how to process each image.  (They are basically just resized and saved in a different folder.)  The script is then supposed to run automatically by PHP calling it using the EXEC command.  This has all worked perfectly for several years now, using 7.0.xxx.   I just installed 7.1.1.-Q16... and I adjusted my script to comply with changes in the Magick commands. The new BAT file PHP creates runs fine with 7.1.1... **IF**_ I manually click it.  But when I ask PHP to run it automatically using EXEC, it doesn't run... no error messages.   Since the only thing different is the updated ImageMagick I am hoping someone can tell me if there is a configuration setting in ImageMagick that I need to change?  Any suggestions are appreciated. Thanks!

**Comment by candidsounds** [comment]
The bat file has lines that are just saying something like "magick "C:\photos\1.jpg[1200x800]" "C:\photossmall\1.jpg"   That's it (I plan to make it a little more complicated eventually).   As I said above, if I click the bat file manually, it works fine...  Any thoughts on why it would work in the older version of ImageMagick but not the latest??   (also, using Apache 2.4.46/PHP 7.4.11)

**Comment by snibgo** [comment]
It could be a permissions issue in PHP or Apache. (I know nothing about PHP or Apache.) I suggest inserting debugging commands in the BAT file to verify whether it is running, and whether `magick` is running, and whether `magick` returns an error.

**Reply by candidsounds** [reply]
Thanks for the reply.  Magick is definitely running correctly because I can manually click the BAT file and it does its "magick"... PHP is not returning any errors, but the EXEC function is a boolean - i.e.- returns TRUE or FALSE... and for some reason it's returning FALSE.  I'm looking through the ini files for php and apache to see if there is something blocking.  Also, Windows itself could be an issue (don't get me started on Windows).  I will report back.

**Comment by candidsounds** [comment, marked as answer]
I got it working, but honestly, I'm not sure why this change had to be made... instead of using PHP "EXEC" function I changed it to the PHP "SYSTEM" function, which is similar, but obviously not exactly the same.  Now the BAT script runs automatically and everything works as expected.  Perhaps this will help someone else in the future.
